docs/apis/subsystems/communication/index.md

```diff
@@ -0,0 +1,244 @@
+---
+title: Communication API
+tags:
+    - communication
+    - provider
+    - Communication provider
+    - Communication room
+    - Communication service
+    - Chat
+documentationDraft: true
+---
+
+Communication API provides access to the messaging system and other communication providers (such as Matrix).
+This API the management of room and members to that room.
+
+Communication API actions its tasks happens asynchronously using ad-hoc tasks. This means that the API functions will return immediately and the action will be
+executed in the background. Communication API has multiple interfaces which acts like a feature of the API. Each of these interface/features have one or more associated
+ad-hoc tasks. These tasks are responsible to action the requested feature from the provider plugin (such as Matrix) informed by the Communication API instance. For developers
+to interact with the API, they must use the public API (communication\api).
+
+Communication API allows to add ad-hoc tasks to the queue to perform actions on the communication providers. This API will not allow any immediate actions to be performed on the
+communication providers. It will only add the tasks to the queue. The exception has been made for deletion of members in case of deleting the user. This is because the user will
+not be available for any checks (capability, role etc.) after deleted.
+
+:::info
+
+This is an experimental feature. To use this feature, you must enable the experimental feature flag for Communication in the site administration.
```

Comment:
I think it would be better to give the exact name of this experimental setting, to help admins find it, something like Enable communication providers (enablecommunicationsubsystem).

Reply:
Done, added the steps for better understanding.